src/components/Floorplan.tsx

```diff
@@ -58,6 +59,75 @@ export const ClickableFloorplan: React.FC<ClickableFloorplanProps> = ({
     }
   }, [imageLoaded, dimensions]);
 
+  // Convert a number of bits (typically megabits) into a string
+  // function megabits(value: number): string {
+  //   return `${(value / 1000000).toFixed(0)}`;
+  // }
+  // Takes a percentage signal strength
+  // returns a rgba() giving a color gradient between red (100%) and blue (0%)
+  function getGradientColor(value: number): string {
+    // Define key color points
+    const colorStops: {
+      value: number;
+      color: [number, number, number, number];
+    }[] = [
+      { value: 100, color: [255, 0, 0, 1] }, // Red
+      { value: 75, color: [255, 255, 0, 1] }, // Yellow
+      { value: 50, color: [0, 255, 0, 1] }, // Green
+      { value: 35, color: [0, 255, 255, 1] }, // Turquoise
+      { value: 0, color: [0, 0, 255, 1] }, // Blue
+      // Color experiment - Green is good, blue is OK, red and yellow are bad
+      // the following values don't quite work...
+      // { value: 100, color: [0, 255, 0, 1] }, // Green
+      // { value: 75, color: [0, 255, 255, 1] }, // Turquoise
+      // { value: 50, color: [0, 0, 255, 1] }, // Blue
+      // { value: 45, color: [255, 255, 255, 1] }, // Grey
+      // { value: 40, color: [255, 255, 0, 1] }, // Yellow
+      // { value: 0, color: [255, 0, 0, 1] }, // Red
+    ];
+
+    // Handle out-of-range values
```

Comment:
Btw. for the final merge, I want us to get rid of these low-value comments (AI code assistants loves to add them).